service/resource/configmap/update.go

```diff
@@ -65,6 +67,10 @@ func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState inte
 			return microerror.Mask(err)
 		}
 
+		// HACK(joe): Wait for the updated configmap to be present in the prometheus container.
+		r.logger.Log("debug", fmt.Sprintf("waiting %v before reloading", r.reloadWaitTime))
+		time.Sleep(r.reloadWaitTime)
```

Comment:
I don't like the time based approach. Why can't we make it event based? Tomorrow it takes 11 seconds for the config map to be propagated while we only wait 10 seconds. What happens then? How can we make this better?